imageio/core/request.py

```diff
@@ -550,6 +552,7 @@ def get_result(self):
         """For internal use. In some situations a write action can have
         a result (bytes data). That is obtained with this function.
         """
+        # Is there a reason to disallow reading multiple times?
```

Comment:
Mostly because the Request object is intended for one-time usage. But the below is also for removing references, to make it easier for the gc and thus lower the chance on memory leaks.

Reply:
😅 Well, currently we are reusing the same request object many times. Mainly because I wasn't aware that it is intended to be a one-time throw-away type object.

`imopen` passes the same Request to each plugin (which is the source of https://github.com/imageio/imageio/issues/642) and the same request is re-used by a plugin when successive `read()` or `write()` calls are performed. If you can think of any problems (beyond https://github.com/imageio/imageio/issues/642) that this could cause, please let me know so that we can track them down to make sure things stay nice and functional.

Reply:
mmm but then its not disallowed, right?

Reply:
> mmm but then its not disallowed, right?

@almarklein Sorry, I don't think I follow. It is not disallowed to read multiple times?

Reply:
I mean, the comment states that is is not allowed to use the request object multiple times, but we *do* use it multiple times now, don't we?

Reply:
@almarklein The comment states that we can't use `Request.get_result` multiple times, and I don't see why this is necessary. There doesn't seem to be anything generally wrong with using the same request object multiple times (apart from #642 ). We have been doing that for the past 8 months or so, doing so passes all of our CI, and so far there are no complaints from downstream 😄.

Reply:
I think the confusion comes from the fact that this looks like a public method, but it was meant for internal use only in a specific worflow where it is only called once :)  Setting the `._result` to `None` is just good practice., but we could do that in `finish()` instead. Nevertheless, might be a good idea to make this method private?

Reply:
> Nevertheless, might be a good idea to make this method private?

Yep, or refactor and stop using it altogether. I like the idea of `Request` being something similar to a fat pointer in C++. I like `Request` to facilitate and manage resource (file/stream/byteobject) access, but I am not so happy with it holding any state beyond that (firstbytes/result/...). That, I would like to see refactored, e.g., into a `memmap` akin to https://github.com/imageio/imageio/issues/616 .